pandas/core/frame.py

```diff
@@ -4991,7 +4992,12 @@ def sort_values(
         if len(by) > 1:
             from pandas.core.sorting import lexsort_indexer
 
-            keys = [self._get_label_or_level_values(x, axis=axis) for x in by]
+            if key is not None:
```

Comment:
this is not a good place for this kind of change at all, rather pass things thru to the functions actually doing the work (lexsort_indexer).

Reply:
Ok. I can add a key to lexsort_indexer and have lexsort_indexer apply the key for `sort_values()`. For `sort_index()` however we'd have to add a key to, among other things, `_sort_levels_monotonic`, `sortlevel`, `is_monotonic_increasing`, and `is_monotonic_decreasing`, which is why it felt much easier to just apply the key function at the beginning.

Reply:
Ok. I can pass most things to lexsort_indexer, but sort_index is complicated for multi-indexes. I feel like the map function should be applied to each multi-index tuple, but the sorting functions are generally applied to the codes and not the values themselves. I can change the sort_values functions to defer work to lexsort_indexer or nargsort, but I think we should keep using index.map(foo) for the sort_index functions.